app/services/users/delete_articles.rb

```diff
@@ -14,6 +14,7 @@ def call(user)
           EdgeCache::BustUser.call(comment.user)
           comment.delete
         end
+        article.discussion_lock&.delete
```

Comment:
This is needed because the `dependent: :destroy` callback is bypassed when we call `article.delete` rather than `article.destroy`. Makes sense.